lib/features/route_map/widgets/bottom_sheet/route_bottom_sheet.dart

```diff
@@ -0,0 +1,83 @@
+import "package:flutter/material.dart";
+import "package:flutter_riverpod/flutter_riverpod.dart";
+
+import "../../../../app/config/ui_config.dart";
+import "../../../../app/l10n/l10n.dart";
+import "../../../../app/theme/app_theme.dart";
+import "../../../../app/theme/color_consts.dart";
+import "../../../../common/data_source/mocks/mock_songs.dart";
+import "../../../../common/providers/bottom_sheet_providers.dart";
+import "../../../../common/widgets/main_action_button.dart";
+import "../../../../common/widgets/map_bottom_sheet.dart";
+import "../../../../common/widgets/secondary_action_button.dart";
+import "../../providers/route_provider.dart";
+import "modals/end_route_modal.dart";
+import "sections/playlist_info_section.dart";
+import "sections/route_info_section.dart";
+
+class RouteBottomSheet extends ConsumerStatefulWidget {
+  const RouteBottomSheet({super.key});
+
+  @override
+  RouteBottomSheetState createState() => RouteBottomSheetState();
+}
+
+class RouteBottomSheetState extends ConsumerState<RouteBottomSheet> {
+  late RouteDetailsOption _chosenOption;
+
+  @override
+  void initState() {
+    super.initState();
+    _chosenOption = RouteDetailsOption.info;
+    Future.microtask(() {
+      ref.read(sheetModeProvider.notifier).state = SheetMode.half;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return MapBottomSheet(
+      button: MainActionButton(
+        text: context.l10n.end_route,
+        backgroundColor: context.colorScheme.error,
+        onPressed: () async {
```

Comment:
as i mentioned before, here, not only you should set the trigger to true, but also the state to hidden, otherwise the sheet won't hide